app/models/action.rb

```diff
@@ -1,6 +1,10 @@
 class Action < ActiveRecord::Base
   belongs_to :page
   belongs_to :action_user
+  after_create :update_page_action_count
 
+  def update_page_action_count
+    self.page.action_count = self.page.actions.count
```

Comment:
> We could move that down to the model

I don't think that's necessary. I'd originally had this code running elsewhere in the chain so it wasn't necessarily protected like it is in `after_create`, so it could just move to being incremented, I think.

One thing I'd double check - make sure that the `action_count` field is always going to be numeric on page create (so, not nil, default 0) in the database. I'm not 100% sure that's true at the moment.